UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -1943,6 +1991,7 @@ protected virtual void ConstructImageFromGLB(GLTFImage image, int imageCacheInde
 			bufferContents.Stream.Position = bufferView.ByteOffset + bufferContents.ChunkOffset;
 			bufferContents.Stream.Read(data, 0, data.Length);
 			texture.LoadImage(data);
+			Progress.TextureLoaded++;
```

Comment:
Check that things aren't getting double counted? Consider making the fields properties with private setters and add functions to increment the values. Then in your functions you can assert that the loaded count does not exceed the total count.

Reply:
I added an assert to LoadSceneAsync as well as asserts to not double load ImageCache values